public/app/plugins/datasource/influxdb/components/queryUtils.ts

```diff
@@ -0,0 +1,57 @@
+import { cloneDeep } from 'lodash';
+import InfluxQueryModel from '../influx_query_model';
+import { InfluxQuery } from '../types';
+
+// FIXME: these functions are a beginning of a refactoring of influx_query_model.ts
+// into a simpler approach with full typescript types.
+// later we should be able to migrate the unit-tests
+// that relate to these functions here, and then perhaps even move the implementation
+// to this place
+
+export function buildRawQuery(query: InfluxQuery): string {
+  const queryCopy = { ...query }; // the query-model mutates the query
```

Comment:
not sure if the query object has nested object, but maybe you want to use `cloneDeep` here as well?